docs/src/4-differentiate-efms.jl

```diff
@@ -0,0 +1,145 @@
+
+        kcat = rid_kcat[rid]
+        parameter_values[Symbol(rid)] = kcat
+
+        d = get!(reaction_isozymes, rid, Dict{String,ParameterIsozyme}())
+        d["isozyme_$i"] = ParameterIsozyme(
+            gene_product_stoichiometry = Dict(grr .=> fill(1.0, size(grr))), # assume subunit stoichiometry of 1 for all isozymes
+            kcat_forward = FastDifferentiation.Node(Symbol(rid)),
+            kcat_reverse = 0.0,
+        )
+        d2 = get!(float_reaction_isozymes, rid, Dict{String,COBREXA.Isozyme}()) #src
+        d2["isozyme_$i"] = COBREXA.Isozyme( #src
+            gene_product_stoichiometry = Dict(grr .=> fill(1.0, size(grr))), #src
+            kcat_forward = kcat, #src
+            kcat_reverse = 0.0, #src
+        ) #src
+    end
+end
+
+
+km = build_kinetic_model(
+    model;
+    reaction_isozymes,
+    gene_product_molar_masses,
+    capacity,
+)
+
+ec_solution, _, _, _ = optimized_constraints_with_parameters(
+    km,
+    parameter_values;
+    objective = km.objective.value,
+    optimizer = Tulip.Optimizer,
+    modifications = [COBREXA.set_optimizer_attribute("IPM_IterationsLimit", 10_000)],
+)
+
+ec_solution.fluxes
+
+ec_solution_cobrexa = enzyme_constrained_flux_balance_analysis( #src
+    model; #src
+    reaction_isozymes = float_reaction_isozymes, #src
+    gene_product_molar_masses, #src
+    capacity, #src
+    optimizer = Tulip.Optimizer, #src
+) #src
+
+@test isapprox(ec_solution.objective, ec_solution_cobrexa.objective; atol = TEST_TOLERANCE) #src
+
+@test any(isapprox.(values(ec_solution.gene_product_amounts), 0, atol=1e-8)) #src
+
+# We have a solution that uses every reaction, and the enzyme capacities are both full.
+# Therefore, we may calculate the EFMs of this solution and direactly differentiate
+# them, with no pruning required.
+
+# ## Calculate EFMs of the optimal solution 
+
+# We need to input the stoichiometric matrix `N`, and a nullspace `K`, into 
+# ElementaryFluxModes.jl
+
+N = AbstractFBCModels.stoichiometry(model)
+
+K = rational_nullspace(Matrix(N))[1]
+
+# Permute the rows of `K` to be in the form `[I;K*]`
+
+order = Int64[]
+rows_done = 1
+for (i, row) in enumerate(eachrow(K))
+    global rows_done
+    rows_done > size(K, 2) && break
+    if row == Matrix(I(size(K, 2)))[rows_done, :]
+        push!(order, i)
+        rows_done += 1
+    end
+end
+append!(order, [i for i = 1:size(K, 1) if i ∉ order])
+K = K[order, :]
+
+# The reaction order of `N` must match that of `K`
+N = N[:, order]
+
+# Run the double description algorithm 
+
+R = DDBinary(N, K)
+E = Matrix(undef, size(R, 1), size(R, 2))
+for (i, r) in enumerate(eachcol(R))
+    non_zero = findall(x -> x != 0, r)
+    flux_ns = rational_nullspace(Matrix(N[:, non_zero]); tol = 1e-14)[1]
+    mode = zeros(size(R, 1))
+    for (j, x) in zip(non_zero, flux_ns)
+        mode[j] = abs(x) < 1e-14 ? 0 : x
+    end
+    E[:, i] = mode
+end
+E = E[invperm(order), :] # put the entries into original reaction order
```

Comment:
this looks like something that should be hidden in the EFM package and hopefully made a bit more efficient (and explained). Will ping on slack.